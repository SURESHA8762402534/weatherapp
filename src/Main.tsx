import React, { FC, useState } from "react";
import { Button, StyleSheet, Text, TextInput, View } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { SafeAreaView } from "react-native-safe-area-context";
import { style } from "./style";

const Main: FC = () => {
  const [name, setName] = useState<string>("");
  const navigation = useNavigation<NativeStackNavigationProp<any>>();

  return (
    <SafeAreaView style={style.container}>
      <SafeAreaView style={style.container}>
        <View style={style.border}>
          <TextInput
            testID="textbox"
            value={name}
            placeholder="Enter Country"
            placeholderTextColor={"gray"}
            onChangeText={(Text) => {
              setName(Text);
            }}
            style={style.input}
          />

          <Button
            testID="btn"
            disabled={!name}
            title="Submit"
            onPress={() => {
              navigation.navigate("details", { name: name });
            }}
          />
        </View>
      </SafeAreaView>
    </SafeAreaView>
  );
};

export default Main;
