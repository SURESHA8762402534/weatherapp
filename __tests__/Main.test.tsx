import { fireEvent, render, screen } from "@testing-library/react-native";
import Main from "../src/Main";

import React from "react";

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

test("test for home page", () => {
  const tree = render(<Main />);
  expect(tree).toMatchSnapshot();
  const input = screen.getByTestId("textbox");
  const btn = screen.getByTestId("btn");
  fireEvent.changeText(input, { target: { value: "test" } });
  fireEvent.press(btn);
});
