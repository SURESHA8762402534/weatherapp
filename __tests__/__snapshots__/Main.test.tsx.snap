// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for home page 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "20px",
      "width": "100%",
    }
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
        "left",
        "right",
        "top",
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "paddingTop": "20px",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "border": "1px solid blue",
          "borderRadius": 10,
          "boxShadow": "0 10px 0 0 solid gray",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": "10px",
          "width": "80%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter Country"
        placeholderTextColor="gray"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderRadius": 8,
            "height": "30px",
            "margin": "20px",
            "width": "100%",
          }
        }
        testID="textbox"
        value=""
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="btn"
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
